test/common/http/conn_manager_impl_test.cc

```diff
@@ -1168,6 +1168,35 @@ TEST_F(HttpConnectionManagerImplTest, FilterAddBodyInline) {
       HeaderMapPtr{new TestHeaderMapImpl{{":status", "200"}}}, true);
 }
 
+TEST_F(HttpConnectionManagerImplTest, FilterClearRouteCache) {
+  setup(false, "");
+
+  EXPECT_CALL(*codec_, dispatch(_)).WillOnce(Invoke([&](Buffer::Instance&) -> void {
+    StreamDecoder* decoder = &conn_manager_->newStream(response_encoder_);
+    HeaderMapPtr headers{new TestHeaderMapImpl{{":authority", "host"}, {":path", "/"}}};
+    decoder->decodeHeaders(std::move(headers), true);
+  }));
+
+  setupFilterChain(2, 2);
+
+  EXPECT_CALL(*route_config_provider_.route_config_, route(_, _)).Times(2);
+
+  EXPECT_CALL(*decoder_filters_[0], decodeHeaders(_, true))
+      .WillOnce(InvokeWithoutArgs([&]() -> FilterHeadersStatus {
+        decoder_filters_[0]->callbacks_->clearRouteCache();
```

Comment:
nit: can you call route() before you call clearRouteCache() ? That would be a slightly better test.